core/src/main/java/org/bitcoinj/crypto/BLSLazySignature.java

```diff
@@ -46,7 +49,8 @@ protected void bitcoinSerializeToStream(OutputStream stream) throws IOException
         lock.lock();
         try {
             if (!isSingatureInitialized && buffer == null) {
-                throw new IOException("singature and buffer are not initialized");
+                log.warn("signature and buffer are not initialized");
+                buffer = invalidSignature.getBuffer();
```

Comment:
Not sure why we are throwing an exception here.  Also not sure how the lazy signature is not initialized or has no buffer.

The idea of a lazy signature is to not initialize a BLS Signature object unless it is needed.  This will save some memory usage, though not as much as a BLSLazyPublicKey.

Reply:
This crash occurs when obtaining the hash value of the InstantSendLock to print it out to the logs.  We probably need to do a little more investigation to figure out why this crash happened.

```
java.io.IOException: singature and buffer are not initialized
 at org.bitcoinj.crypto.BLSLazySignature.bitcoinSerializeToStream(BLSLazySignature.java:49)
 at org.bitcoinj.core.Message.bitcoinSerialize(Message.java:268)
 at org.bitcoinj.quorums.InstantSendLock.bitcoinSerializeToStream(InstantSendLock.java:99)
 at org.bitcoinj.quorums.InstantSendLock.getHash(InstantSendLock.java:119)
 at org.bitcoinj.quorums.InstantSendManager.processInstantSendLock(InstantSendManager.java:589)
 at org.bitcoinj.quorums.InstantSendManager.processPendingInstantSendLocks(InstantSendManager.java:529)
 at org.bitcoinj.quorums.LLMQBackgroundThread.run(LLMQBackgroundThread.java:36)
```